projectq/libs/revkit/_control_function_test.py

```diff
@@ -0,0 +1,91 @@
+#   Copyright 2017 ProjectQ-Framework (www.projectq.ch)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Tests for libs.revkit._control_function."""
+
+import pytest
+
+from projectq.types import Qubit
+from projectq import MainEngine
+from projectq.cengines import DummyEngine
+
+from projectq.libs.revkit import ControlFunctionOracle
+
+
+# run this test only if RevKit Python module can be loaded
+revkit = pytest.importorskip('revkit')
+
+
+def test_control_function_majority():
+    saving_backend = DummyEngine(save_commands=True)
+    main_engine = MainEngine(backend=saving_backend,
+                             engine_list=[DummyEngine()])
+
+    qubit0 = Qubit(main_engine, 0)
```

Comment:
Can we change this to `qubit0 = main_engine.allocate_qubit()`? This will, in addition, send allocate gates so the command count below needs to be adapted but it is how qubits should be allocated.